src/OpenSage.Tools.BigEditor/UI/MainForm.cs

```diff
@@ -0,0 +1,304 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Numerics;
+using ImGuiNET;
+using OpenSage.Data.Big;
+using OpenSage.Tools.BigEditor.Util;
+using Veldrid.Sdl2;
+
+namespace OpenSage.Tools.BigEditor.UI
+{
+    internal sealed class MainForm : DisposableBase
+    {
+        private BigArchive _bigArchive;
+        private readonly List<BigArchiveEntry> _files;
+        private int _currentFile;
+        private string _currentFileText;
+
+        private byte[] _searchTextBuffer;
+        private byte[] _filePathBuffer;
+        private string _searchText;
+
+        public MainForm()
+        {
+            _files = new List<BigArchiveEntry>();
+            OpenBigFile(null);
+        }
+
+        private void UpdateSearch(string searchText)
+        {
+            searchText = ImGuiUtility.TrimToNullByte(searchText);
+
+            if (searchText == _searchText)
+            {
+                return;
+            }
+
+            _searchText = searchText;
+
+            _files.Clear();
+
+            if (_bigArchive == null)
+            {
+                return;
+            }
+
+            if (string.IsNullOrWhiteSpace(_searchText))
+            {
+                _files.AddRange(_bigArchive.Entries.OrderBy(x => x.FullName));
+            }
+            else
+            {
+                _files.AddRange(_bigArchive.Entries
+                    .Where(entry => entry.FullName.IndexOf(_searchText, StringComparison.OrdinalIgnoreCase) >= 0)
+                    .OrderBy(x => x.FullName));
+            }
+        }
+
+        public void Draw(Sdl2Window window)
+        {
+            ImGui.SetNextWindowPos(Vector2.Zero, ImGuiCond.Always, Vector2.Zero);
+            ImGui.SetNextWindowSize(new Vector2(window.Width, window.Height), ImGuiCond.Always);
+
+            ImGui.PushStyleVar(ImGuiStyleVar.WindowRounding, 0);
+            bool open = false;
+            ImGui.Begin("OpenSAGE Big Editor", ref open, ImGuiWindowFlags.MenuBar | ImGuiWindowFlags.NoTitleBar);
```

Comment:
Add `ImGuiWindowFlags.NoResize` flag here to remove the extraneous resize triangle from the bottom right corner.